pypeit/core/coadd.py

```diff
@@ -2920,11 +2924,21 @@ def compute_coadd2d(ref_trace_stack, sciimg_stack, sciivar_stack, skymodel_stack
         dspat_img_fake = spat_img_coadd + dspat_mid[0]
         dspat[np.invert(outmask)] = dspat_img_fake[np.invert(outmask)]
 
+    # update maskdef_objpos and maskdef_slitcen with the new value in the new slit
```

Comment:
This sequence should only be executed if maskef_dict is passed in. You can leave the hook for masked_designtab in the return dictionary, and set it to None if no maskdef_dict was passed in.